plugins/qmlsupport/qmllistpropertyadaptor.cpp

```diff
@@ -60,9 +62,13 @@ PropertyData QmlListPropertyAdaptor::propertyData(int index) const
     auto var = object().variant(); // we need to keep that alive for the runtime of this method
     QQmlListProperty<QObject> *prop
         = reinterpret_cast<QQmlListProperty<QObject> *>(const_cast<void *>(var.data()));
+    if (!prop->count)
+        return pd;
     if (!prop || index >= prop->count(prop))
```

Comment:
Checking for `prod != nullptr` after dereferencing it (see line 65) makes no sense. You need to reorder those statements.

```
if (!prop)
    return pd;

if (!prop->count || index >= ...)
    return pd;
```